HumanDynamicsEstimationLibrary/include/InverseVelocityKinematics.hpp

```diff
@@ -21,119 +21,120 @@
-                                          iDynTree::MatrixDynSize customConstraintMatrix,
-                                          double k_u,
-                                          double k_l);
-
-    bool setGeneralJointVelocityConstraints(double jointVelocityLimit);
-
-    bool setGeneralJointsUpperLowerConstraints(iDynTree::VectorDynSize jointUpperLimits,
-                                               iDynTree::VectorDynSize jointLowerLimits);
-
-    bool updateTarget(std::string linkName,
-                      iDynTree::Vector3 linearVelocity,
-                      iDynTree::Vector3 angularVelocity,
-                      double linearWeight = 1.0,
-                      double angularWeight = 1.0);
-    bool updateTarget(std::string linkName,
-                      iDynTree::Twist twist,
-                      double linearWeight = 1.0,
-                      double angularWeight = 1.0);
-    bool updateLinearVelocityTarget(std::string linkName,
-                                    iDynTree::Vector3 linearVelocity,
-                                    double linearWeight = 1.0);
-    bool updateAngularVelocityTarget(std::string linkName,
-                                     iDynTree::Vector3 angularVelocity,
-                                     double angularWeight = 1.0);
-
-    bool getVelocitySolution(iDynTree::Twist& baseVelocity,
-                             iDynTree::VectorDynSize& jointsVelocity);
-    bool getJointsVelocitySolution(iDynTree::VectorDynSize& jointsVelocity);
-    bool getBaseVelocitySolution(iDynTree::Twist& baseVelocity);
-    bool getBaseVelocitySolution(iDynTree::Vector3& linearVelocity,
-                                 iDynTree::Vector3& angularVelocity);
-
-    bool solve();
-    void clearProblem();
-};
-
-}
+    protected:
+        class impl;
+        std::unique_ptr<impl> pImpl;
+
+    public:
+        typedef enum
+        {
+            QP,
+            moorePenrose,
+            completeOrthogonalDecomposition,
+            leastSquare,
+            choleskyDecomposition,
+            sparseCholeskyDecomposition,
+            robustCholeskyDecomposition,
+            sparseRobustCholeskyDecomposition,
+        } InverseVelocityKinematicsResolutionMode;
+
+        InverseVelocityKinematics();
+        ~InverseVelocityKinematics();
+
+        bool setModel(iDynTree::Model model);
+        bool setFloatingBaseOnFrameNamed(std::string floatingBaseFrameName);
+        bool setResolutionMode(InverseVelocityKinematicsResolutionMode resolutionMode);
+        bool setResolutionMode(std::string resolutionModeName);
+        void setRegularization(double regularizationWeight);
+
+        bool addTarget(std::string linkName,
+                       iDynTree::Vector3 linearVelocity,
+                       iDynTree::Vector3 angularVelocity,
+                       double linearWeight = 1.0,
+                       double angularWeight = 1.0);
```

Comment:
The arguments of all these methods could become `const <type>&`. You can omit the `&` for built-in data types like double.